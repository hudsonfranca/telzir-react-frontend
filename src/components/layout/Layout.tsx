import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Grid from './styles';
import Header from '../Header/Header';
import Main from '../Main/Main';
import Routes from '../../Routes';
import Footer from '../Footer/Footer';
import SideDrawerProvider from '../context/SideDrawerContext';
import SideDrawer from '../SideDrawer/SideDrawer';

const Layout: React.FC = () => {
  return (
    <BrowserRouter>
      <Grid data-test="layoutGrid">
        <SideDrawerProvider data-test="sidedrawerProvider">
          <Header data-test="header" />
          <SideDrawer data-test="sideDrawer" />
        </SideDrawerProvider>
        <Main data-test="main">
          <Routes data-test="routes" />
        </Main>
        <Footer data-test="footer" />
      </Grid>
    </BrowserRouter>
  );
};

export default Layout;
