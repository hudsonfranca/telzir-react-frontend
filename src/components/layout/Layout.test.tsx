import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import renderer from 'react-test-renderer';
import Layout from './Layout';
import 'jest-styled-components';

let wrapper: ShallowWrapper;

beforeEach(() => {
  wrapper = shallow(<Layout />);
});

describe('<Layout /> redering', () => {
  it('Should render without errors', () => {
    const layout = renderer.create(<Layout />).toJSON();

    expect(layout).toMatchSnapshot();
  });

  it('Should render 1 layoutGrid', () => {
    expect(wrapper.find(`[data-test='layoutGrid']`).length).toEqual(1);
  });

  it('Should render 1 sidedrawerProvider', () => {
    expect(wrapper.find(`[data-test='sidedrawerProvider']`).length).toEqual(1);
  });

  it('Should render 1 header', () => {
    expect(wrapper.find(`[data-test='header']`).length).toEqual(1);
  });

  it('Should render 1 sideDrawer', () => {
    expect(wrapper.find(`[data-test='sideDrawer']`).length).toEqual(1);
  });

  it('Should render 1 main', () => {
    expect(wrapper.find(`[data-test='main']`).length).toEqual(1);
  });

  it('Should render 1 routes', () => {
    expect(wrapper.find(`[data-test='routes']`).length).toEqual(1);
  });

  it('Should render 1 footer', () => {
    expect(wrapper.find(`[data-test='footer']`).length).toEqual(1);
  });
});
