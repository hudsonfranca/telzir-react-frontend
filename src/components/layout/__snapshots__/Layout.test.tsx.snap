// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> redering Should render without errors 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto auto 170px;
  grid-template-areas: 'header' 'main' 'footer';
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  grid-area: header;
  width: 100%;
  height: 60px;
  background: rgba(11,12,16,0.5);
  position: fixed;
  z-index: 20;
  padding-left: 20px;
  padding-right: 20px;
}

.c1 ul {
  width: 100%;
  height: 100%;
  font-size: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-left: 20px;
  padding-right: 20px;
}

.c1 ul img {
  cursor: pointer;
}

.c1 ul li {
  width: 100px;
  height: 100%;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 ul li a {
  color: var(--white);
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

.c1 ul li a:hover,
.c1 ul li a:active {
  color: var(--quaternary);
}

.c4 {
  width: 50px;
  height: 50px;
  padding: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c4 div {
  width: 100%;
  height: 3px;
  background-color: var(--white);
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
  border-radius: 10px;
}

.c4 div:first-child {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c4 div:nth-child(2) {
  opacity: 1;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.c4 div:nth-child(3) {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c6 {
  grid-area: main;
  position: relative;
}

.c7 {
  width: 100%;
  min-height: 100vh;
}

.c8 {
  width: 100%;
  height: 100vh;
  position: relative;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--white);
}

.c8 h1 {
  font-size: 4rem;
  padding-bottom: 20px;
  width: 80%;
  text-align: center;
}

.c8 p {
  font-size: 2rem;
  width: 80%;
  text-align: center;
}

.c8::after {
  content: '';
  width: 100%;
  height: 100%;
  background: rgba(11,12,16,0.5);
  position: absolute;
  top: 0;
  z-index: -1;
}

.c11 {
  width: 100%;
  grid-area: r1;
  height: 90%;
  background-image: url(phone.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  position: relative;
  z-index: 1;
}

.c11::after {
  content: '';
  width: 100%;
  height: 100%;
  background: rgba(11,12,16,0.7);
  position: absolute;
  top: 0;
  z-index: -1;
}

.c14 {
  width: 100%;
  grid-area: r2;
  height: 90%;
  background-image: url(phone.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  position: relative;
  z-index: 1;
}

.c14::after {
  content: '';
  width: 100%;
  height: 100%;
  background: rgba(11,12,16,0.7);
  position: absolute;
  top: 0;
  z-index: -1;
}

.c15 {
  width: 100%;
  grid-area: r3;
  height: 90%;
  background-image: url(phone.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  color: var(--white);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  position: relative;
  z-index: 1;
}

.c15::after {
  content: '';
  width: 100%;
  height: 100%;
  background: rgba(11,12,16,0.7);
  position: absolute;
  top: 0;
  z-index: -1;
}

.c10 {
  color: var(--white);
  font-size: 4rem;
  height: 100%;
  width: 100%;
  grid-area: title;
  text-align: center;
  padding: 50px;
}

.c12 {
  font-size: 3rem;
  height: 100%;
  margin-bottom: 10%;
  text-align: center;
}

.c13 {
  font-size: 2rem;
  height: 100%;
  margin-bottom: 30px;
  text-align: center;
}

.c9 {
  width: 100%;
  min-height: 100vh;
  background-color: var(--quaternary);
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: auto 1fr;
  grid-template-areas: 'title title title' 'r1 r2 r3';
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 30px;
  gap: 20px;
}

.c16 {
  width: 100%;
  min-height: 100vh;
  background-color: var(--secondary);
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-template-rows: auto 1fr;
  grid-template-areas: 'title title title' 'r1 r2 r3';
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 30px;
  gap: 20px;
}

.c17 {
  grid-area: r1;
  width: 100%;
  height: 90%;
  background-color: var(--primary);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px;
}

.c17 strong a {
  color: var(--primary);
}

.c17 strong a:hover,
.c17 strong a:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: #777;
  font-size: 0.8em;
  margin: 0.5em 0;
  position: relative;
  width: 80%;
  height: 75px;
}

.c21 {
  width: 100%;
  height: 40px;
  border-radius: 5px;
  border: none;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  font-size: 1.6rem;
  background-color: var(--white);
  padding-left: 5px;
}

.c21:focus {
  outline: none !important;
  border: none;
  box-shadow: 0 0 9px var(--hover1);
}

.c20 {
  width: 100%;
  height: 40px;
  border: none;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  font-size: 1.6rem;
  background-color: var(--white);
  padding-left: 5px;
}

.c20:focus {
  outline: none !important;
  border: none;
  box-shadow: 0 0 9px var(--quaternary);
}

.c19 {
  color: var(--white);
  font-size: 1.4rem;
  margin-bottom: 5px;
}

.c22 {
  border-radius: 10px;
  width: 80%;
  height: 40px;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  color: var(--primary);
  font-size: 1.8rem;
  background-color: var(--quaternary);
}

.c23 {
  grid-area: footer;
  width: 100%;
  height: 100%;
  background-color: var(--primary);
  font-size: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  position: relative;
}

.c23 small {
  color: var(--white);
  font-size: 1.6rem;
}

.c23::before {
  content: '';
  width: 100%;
  height: 5px;
  position: absolute;
  top: 0;
  background-image: linear-gradient( to right, var(--primary), var(--quaternary), var(--quaternary), var(--primary) );
}

.c5 {
  width: 100vw;
  height: 100vh;
  position: fixed;
  top: 0;
  left: 0;
  background-color: var(--primary);
  -webkit-transition: 0.8s ease-in-out;
  transition: 0.8s ease-in-out;
  z-index: 20;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: auto 2fr;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-transform: translate3d(-100%,0,0);
  -ms-transform: translate3d(-100%,0,0);
  transform: translate3d(-100%,0,0);
}

.c5 div {
  margin: 3px 0 0 15px;
}

.c5 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
  color: var(--white);
  font-size: 2rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 ul li {
  padding: 20px;
  width: 90%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 ul li:hover,
.c5 ul li:active {
  background-color: var(--secondary);
}

.c5 ul li a {
  color: var(--white);
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

@media (max-width:550px) {
  .c1 ul {
    -webkit-box-pack: space-around;
    -webkit-justify-content: space-around;
    -ms-flex-pack: space-around;
    justify-content: space-around;
  }

  .c1 ul li {
    display: none;
  }
}

@media (min-width:550px) {
  .c1 ul .c2 {
    display: none;
  }
}

@media (max-width:767px) {
  .c9 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:767px) {
  .c16 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:736px) {
  .c23 ul {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (min-width:1000px) {
  .c5 {
    display: none;
  }
}

<div
  className="c0"
  data-test="layoutGrid"
>
  <header
    className="c1"
    data-test="headerContainer"
  >
    <ul
      data-test="headerList"
    >
      <div
        className="c2 c3"
        data-test="headerHbContainer"
      >
        <div
          className="c4"
          data-test="hamburguerButtonContainer"
          onClick={[Function]}
          open={false}
        >
          <div
            data-test="hamburguerLine"
          />
          <div
            data-test="hamburguerLine"
          />
          <div
            data-test="hamburguerLine"
          />
        </div>
      </div>
      <img
        alt="Logo"
        data-test="headerLogo"
        src="Logo1.0.png"
      />
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
           Contact
        </a>
      </li>
    </ul>
  </header>
  <div
    className="c5"
    data-test="sidedrawerContainer"
  >
    <div
      className="c4"
      data-test="hamburguerButtonContainer"
      onClick={[Function]}
      open={false}
    >
      <div
        data-test="hamburguerLine"
      />
      <div
        data-test="hamburguerLine"
      />
      <div
        data-test="hamburguerLine"
      />
    </div>
    <ul
      data-test="menu"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
  </div>
  <main
    className="c6"
    data-test="mainContainer"
  >
    <div
      className="c7"
      data-test="homeContainer"
    >
      <div
        className="react-parallax "
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <img
          alt=""
          className="react-parallax-bgimage"
          src="home.jpg"
          style={
            Object {
              "MozBackfaceVisibility": "hidden",
              "MsBackfaceVisibility": "hidden",
              "WebkitBackfaceVisibility": "hidden",
              "WebkitTransform": "translate3d(-50%, 0, 0)",
              "WebkitTransformStyle": "preserve-3d",
              "left": "50%",
              "position": "absolute",
              "transform": "translate3d(-50%, 0, 0)",
            }
          }
        />
        <div
          className="react-parallax-content"
          style={
            Object {
              "position": "relative",
            }
          }
        >
          <section
            className="c8"
            data-test="homeSectionOne"
          >
            <h1>
               Cofira os produtos e serviços da Telzir.
            </h1>
            <p>
              Lorem, ipsum dolor sit amet consectetur adipisicing elit. Animi exercitationem provident facilis nulla nam vero quod est magnam dolorum itaque ipsum, eligendi quam labore optio explicabo? Ad molestiae laboriosam quam?
            </p>
          </section>
        </div>
      </div>
      <section
        className="c9"
        data-test="homeSectionTwo"
      >
        <p
          className="c10"
          data-test="homeSectionTwoTitle"
        >
          Nossos melhores planos
        </p>
        <div
          className="c11 fadeIn"
          data-test="homeSectionTwoCard"
        >
          <p
            className="c12"
          >
            Fale Mais 30
          </p>
          <p
            className="c13"
          >
            Fale até 30 minutos sem pagar nada
          </p>
        </div>
        <div
          className="c14 fadeIn"
          data-test="homeSectionTwoCard"
        >
          <p
            className="c12"
          >
            Fale Mais 60
          </p>
          <p
            className="c13"
          >
            Fale até 60 minutos sem pagar nada
          </p>
        </div>
        <div
          className="c15 fadeIn"
          data-test="homeSectionTwoCard"
        >
          <p
            className="c12"
          >
            Fale Mais 120
          </p>
          <p
            className="c13"
          >
            Fale até 120 minutos sem pagar nada
          </p>
        </div>
      </section>
      <section
        className="c16"
        data-test="homeSectionThree"
      >
        <p
          className="c10"
        >
          Compare os nosos Planos
        </p>
        <form
          className="c17"
          data-test="homeSectionThreeForm"
          onSubmit={[Function]}
        >
          <label
            className="c18"
          >
            <label
              className="c19"
              htmlFor="priceId"
            >
              Selecione a origem e destino
            </label>
            <select
              className="c20"
              id="priceId"
              name="priceId"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Origem e Destino
              </option>
            </select>
          </label>
          <label
            className="c18"
          >
            <label
              className="c19"
              htmlFor="planId"
            >
               Selecione um plano
            </label>
            <select
              className="c20"
              id="planId"
              name="planId"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Plano
              </option>
            </select>
          </label>
          <label
            className="c18"
          >
            <label
              className="c19"
              htmlFor="minutes"
            >
              Quantidade de minutos
            </label>
            <input
              className="c21"
              id="minutes"
              max="1000"
              min="1"
              name="minutes"
              onChange={[Function]}
              type="number"
              value=""
            />
          </label>
          <button
            className="c22"
          >
            Comparar
          </button>
        </form>
      </section>
    </div>
  </main>
  <div
    className="c23"
    data-test="container"
  >
    <img
      alt="Logo"
      src="Logo1.0.png"
    />
    <small>
      ©
      Telzir 2020  All rights reserved.
    </small>
  </div>
</div>
`;
